pyproject.toml

```diff
@@ -58,3 +58,10 @@ packages = ["multipart"]
 
 [tool.hatch.build.targets.sdist]
 include = ["/multipart", "/tests"]
+
+[tool.pyright]
+include = ["multipart", "tests"]
+exclude = [
+    "**/__pycache__",
+]
+reportUndefinedVariable = false # TODO: required because pyright does not work with __future__.annotations
```

Comment:
It does work... What do you mean by this?

Reply:
Hi @Kludex!

What I mean by this is that Pyright will raise "false positive" `undefinedVariable` warnings.

```
➜  ~ pyright --version
pyright 1.1.349

➜  ~ cat kludex.py
from __future__ import annotations

def foo(s: str) -> List[str]: # Gonna get a Pyright complain here
    return s.split()

➜  ~ pyright kludex.py
/Users/eltbus/kludex.py
  /Users/eltbus/kludex.py:3:20 - error: "List" is not defined (reportUndefinedVariable)
1 error, 0 warnings, 0 informations
```

Reply:
Pyright is right there... You need to use `list` instead of `List`. 🤔 

`List` should be imported if you want to use it anyway.

Reply:
Oh, I believed this would not work for Python3.7, but it indeed works.
Nonetheless pyright will also complain for `Optional`, `Union`, `Callable`, unless explicitly imported from typing.

Reply:
I mean, that's a python thing, not a Pyright... If the class is not present, and is not built-in, it needs to be imported.

Instead of `Optional[T]`, use `T | None`; instead of `Union[T, K]`, use `T | K`; `Callable` just needs to be imported.

Reply:
Isn't that Python3.10 syntax?

Reply:
If you use `__future__.annotations`, what is on the type annotations doesn't matter for runtime.